File: go-client-sdk/pkg/models/shared/systemobjectpushoperationpagedresponsemodel.go
package shared

// SystemObjectPushOperationPagedResponseModel
// Used to represent what can be returned by an endpoint that supports paging.
// Usable with the [ProducesResponseType] attribute on a controller action.
type SystemObjectPushOperationPagedResponseModel struct {
	Links        *SystemObjectPushOperationPagedResponseLinksModel `json:"_links,omitempty"`
	PageNumber   *int                                              `json:"pageNumber,omitempty"`
	PageSize     *int                                              `json:"pageSize,omitempty"`
	Results      []SystemObjectPushOperation                       `json:"results,omitempty"`
	TotalResults *int                                              `json:"totalResults,omitempty"`
}

File: python-client-sdk/src/sdk/models/shared/system_objectpushoperationpagedresponsemodel.py
import dataclasses
from ..shared import system_objectpushoperation as shared_system_objectpushoperation
from ..shared import system_objectpushoperationpagedresponselinksmodel as shared_system_objectpushoperationpagedresponselinksmodel
from dataclasses_json import dataclass_json
from sdk import utils
from typing import Optional


@dataclass_json
@dataclasses.dataclass
class SystemObjectPushOperationPagedResponseModel:
    r"""SystemObjectPushOperationPagedResponseModel
    Used to represent what can be returned by an endpoint that supports paging.
    Usable with the [ProducesResponseType] attribute on a controller action.
    """
    
    links: Optional[shared_system_objectpushoperationpagedresponselinksmodel.SystemObjectPushOperationPagedResponseLinksModel] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('_links') }})
    page_number: Optional[int] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('pageNumber') }})
    page_size: Optional[int] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('pageSize') }})
    results: Optional[list[shared_system_objectpushoperation.SystemObjectPushOperation]] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('results') }})
    total_results: Optional[int] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('totalResults') }})
    